### DISCUSSION TITLE
Starting Bonuses in Campaigns

### DISCUSSION THREAD

**Discussion by zenseii** [ihhub/fheroes2#4541] [General]
Opening this discussion as a general one regarding the starting bonuses for campaigns, both as a way to share info and to share opinions on changes. @pflugs30 mentioned that it might be better to change the use from enums to classes for them.

Anyways, here's a list of all starting bonuses that are artifacts:

<details>
<summary>All Unique Campaign Starting Bonus Artifacts A-Z for Heroes 2 TSW and PoL (click to drop down)</summary>
<br>

Ballista
Black Pearl
Breastplate
Caster's Bracelet
Defender Helm
Dragon Sword
Fizbin Medal
Foremost Scroll
Gauntlets
Hideous Mask
Mage's Ring
Major Scroll
Medal of Honor
Medal of Valor
Minor Scroll
Nomad Boots
Power Axe
Spiked Shield
Stealth Shield
Tax Lien
Thunder Mace
Traveler's Boots
White Pearl

</details>

### Complete Tables of Starting Bonuses of all Campaigns ("unreadable" on phones):
_NB.: There are always 3 bonuses per scenario. All bonuses are reported as they appear in-game._

| Roland (Good) |  |   Archibald (Evil) |  | The Price of Loyalty |  |
| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |
| _**Scenario #**_ | _**Starting Bonus**_ | _**Scenario #**_ | _**Starting Bonus**_ | _**Scenario #**_ | _**Starting Bonus**_ |
| **Scenario 1** | 2000 Gold  |  **Scenario 1** | 2000 Gold  | **Scenario 1** | Medal of Valor  |
|  | Thunder Mace  | | Mage's Ring  | | Stealth Shield |
|   | Gauntlets |   | Minor Scroll |   | Minor Scroll |
|  **Scenarios 2-5 & 8** | Wizard  |**Scenarios 2, 4-7 & 9** | Necromancer  | **Scenario 2** | White Pearl  |
|  | Sorceress  |  | Warlock  | | Ballista  |
|   | Knight  |  |  Barbarian |  | 2000 Gold  |
|  **Scenario 6** | Mirror Image  |  **Scenario 3** | 2000 Gold  | **Scenario 3** | Mage's Ring  |
|  | Summon Earth  |  | Mass Curse  |  | Major Scroll  |
|   | Resurrect  |  | Defender Helm  |   | Adv. Logistics  |
|  **Scenario 7** | Black Pearl  |  **Scenario 8** | Basic Logistics  |  **Scenario 4** | Medal of Honor  |
|  | Dragon Sword  |  | Power Axe  | | Caster's Bracelet  |
|   | Breastplate  |   | White Pearl  |   | Defender Helm  |
|  **Scenario 9** | 20 Crystal  | **Scenario 10** | Black Pearl  | **Scenario 5** | 2000 Gold  |
|  | 20 Gems  | | Dragon Sword  | | Power Axe  |
|   | 20 Mercury  |  | Breast Plate  |   | Defense Skill +2  |
|  **Scenario 10** | Tax Lien  |   **Scenario 11** | Tax Lien  | **Scenario 6** | 2000 Gold  |
|  | Hideous Mask  |  | Hideous Mask  | | 3 Crusaders  | 
|   | Fizbin Medal  |   | Fizbin Medal  |  | Traveler's Boots  |
|  |   |  |  |  **Scenario 7** | 10 Sulfur  |
|  |   |  |  |   | Power Axe  |
|  |   |  |  |     | Animate Dead  |
|  |   |  |  |   **Scenario 8** | View Heroes  |
|  |   |  |  |   | 5 Magi  |
|  |   |  |  |    | Advanced Estates  |

| Voyage Home |  | Wizard's Isle |  | Descendants |  | 
| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |
| _**Scenario #**_ | _**Starting Bonus**_ | _**Scenario #**_ | _**Starting Bonus**_ | _**Scenario #**_ | _**Starting Bonus**_ |
| **Scenario 1** | Nomad Boots | **Scenario 1** | 2000 Gold | **Scenario 1** | 1000 Gold  |
|  | 500 Gold |   | Power Axe |  | 50 Wood |
|   | Basic Scouting |  | Stealth Shield |   | 10 Crystal |
|  **Scenario 2** | Basic Pathfinding  |**Scenario 2** | Mage's Ring  |  **Scenario 2** | Attack Skill +1 |
|  | Basic Navigation  |  | Foremost Scroll  |  | Defense Skill +1  |
|   | 15 Sulfur  |  | Fireblast  |  | 1000 Gold  |
|  **Scenario 3** | Basic Archery  | **Scenario 3** | Mass Haste  |  **Scenario 3** | Basic Logistics  |
|  | 3000 Gold  |  | Summon Earth  |  | Steelskin  |
|   | Attack Skill +1  |   | Chain Lightning  |   | Mage's Ring  |
|  **Scenario 4** | Bas. Necromancy  |  **Scenario 4** | 5 Wood  |  **Scenario 4** | Basic Logistics |
|  | Basic Logistics  |  | 5 Ore  |  | Basic Scouting  |
|   | 2000 Gold  |   | 1000 Gold  |   | Power Axe  |
| | | | | **Scenario 5** | Basic Luck  |
| | | | |   | Basic Leadership  |
| | | | |   | Spiked Shield  |
| | | | |  **Scenario 6** | -1000 Gold  |
| | | | |   | -10 Wood  |
| | | | |   | -2 Mercury  |
| | | | |   **Scenario 7** | 1000 Gold  |
| | | | |  | Basic Scouting  |
| | | | |   | 20 Wood  |
| | | | |   **Scenario 8** | Basic Leadership  |
| | | | |   | Basic Archery  |
| | | | |   | Spell Power +1  |

**Comment by pflugs30** [comment]
> @pflugs30 mentioned that it might be better to change the use from enums to classes for them.

Please see the discussion at #4517 for a related conversation.

**Comment by pflugs30** [comment]
See also #4421

**Comment by zenseii** [comment]
@pflugs30 I finished adding the complete table of starting bonuses in the first post. Sadly no colors exist on github so the readability isn't optimal.

One thing I noticed is that OG has the short forms: `Bas. Necromacy` in scenario 4 of Voyage Home, and `Adv. Logistics` in scenario 3 of The Price of Loyalty. Not sure how these appear in fheroes2.

Edit: Right now secondary skill bonuses don't have their own context, so they appear in their full length name, and therefore are too long. The issue of contexts still remains in other words.
